server/src/main/java/org/elasticsearch/gateway/PersistedClusterStateService.java

```diff
@@ -1008,7 +1009,7 @@ void commit(long currentTerm, long lastAcceptedVersion, Version oldestIndexVersi
         }
 
         private boolean assertOnCommit() {
-            if (assertOnCommit != null /* TODO && Randomness.get().nextInt(100) == 0 */) {
+            if (assertOnCommit != null && Randomness.get().nextInt(100) == 0) {
```

Comment:
I wonder if this should be an (internal) setting? This way it could be easier to control in tests as well as disable in production if this affects somebody